src/Stratis.Bitcoin/Utilities/DBreezeSerializer.cs

```diff
@@ -91,25 +90,18 @@ public T Deserialize<T>(byte[] bytes)
         /// <returns>Deserialized object.</returns>
         public object Deserialize(byte[] bytes, Type type)
         {
-            if (type == typeof(Coins))
-            {
-                var coin = new Coins();
-                coin.ReadWrite(bytes, this.Network.Consensus.ConsensusFactory);
-                return coin;
-            }
-
             if (type == typeof(BlockHeader))
             {
                 BlockHeader header = this.Network.Consensus.ConsensusFactory.CreateBlockHeader();
                 header.ReadWrite(bytes, this.Network.Consensus.ConsensusFactory);
                 return header;
             }
 
-            if (type == typeof(RewindData))
+            if (type == typeof(Transaction))
```

Comment:
This is odd did we not serialize trx before?

Reply:
This method is specifying how we deserialize objects and seeing as Transaction has to always be created by the network's consenus factory it's more correct to specifically specify how transactions should be deserialized.

Reply:
This ensures that when we read transactions as root objects (i.e. not embedded in Block) from the DBreeze database that the appropriate transaction type is created - i.e.. Transaction or PosTransaction.

Reply:
@quantumagi ah I see, that's because we actually don't do it correctly, we should let the serializer create the instance , same problem with Block and Header